lib/AST/ASTScopeLookup.cpp

```diff
@@ -703,42 +703,24 @@ lookupEnclosingABIAttributeScope(SourceFile *sourceFile, SourceLoc loc) {
   return nullptr;
 }
 
-static std::pair<CatchNode, const BraceStmtScope *>
-getCatchNodeBody(const ASTScopeImpl *scope, CatchNode node) {
-  const auto &children = scope->getChildren();
-  if (children.empty())
-    return { CatchNode(), nullptr };
-
-  auto stmt = dyn_cast<BraceStmtScope>(children[0]);
-  if (!stmt || stmt->getStmt()->empty())
```

Comment:
@DougGregor It didn't appear to be necessary to preserve the empty check here, do you remember what case that was handling? Happy to add it back

Reply:
I do not remember what case it was handling. It may have been me being defensive around the `dyn_cast`.